lib/ProductOpener/Tags.pm

```diff
@@ -782,11 +782,101 @@ sub remove_plurals($$) {
 
 
 
+=head2 sanitize_taxonomy_line( $line )
+
+Sanitize a taxonomy line before processing
+
+=head3 Arguments
+
+=head4 str $line - the line read from the file
+
+=cut
+sub sanitize_taxonomy_line($)
+{
+	my $line = shift;
+ 	
+	chomp($line);
+
+	$line =~ s/’/'/g;  # normalize quotes
+
+	# assume commas between numbers are part of the name
+	# e.g. en:2-Bromo-2-Nitropropane-1,3-Diol, Bronopol
+	# replace by a lower comma ‚
+
+	$line =~ s/(\d),(\d)/$1‚$2/g;
+
+	# replace escaped comma \, by a lower comma ‚
+	$line =~ s/\\,/‚/g;
+
+	# remove parenthesis for roman numerals
+	# fr:E333(iii), Citrate tricalcique
+	# -> E333iii
+
+	$line =~ s/\(((i|v|x)+)\)/$1/i;
+
+	# strip spaces at end of line
+	$line =~ s/\s+$//;
+
+	return $line
+}
+
+
+=head2 search_tag_c( $tag, $synonyms, $tagtype, $lc, $warning )
+
+Search for "current tag" (tag at start of line) for a given tag
+
+=head3 Arguments
+
+=head4 str $tag - tag string for which we search
+
+=head4 reference to hash map $synonyms - ref to $synonyms for $tagtype
+
+=head4 str $tagtype - tag type
+
+=head4 str $lc - language
+
+=head4 str $warning
+
+An optional prefix to display errors if we had to use stopwords / plurals.
+
+If empty, no warning will be displayed.
+
+=head3 return str - found current tagid or undef
+
+=cut
+sub search_tag_c($$$$$)
+{
+	my $tag = shift;
+	my $synonyms = shift;
```

Comment:
The naming convention we have is to suffix all references with _ref, so it would be $synonyms_ref.